integration_tests/features/support/steps.js

```diff
@@ -2569,28 +2543,34 @@ Then(
         );
         const isTransactionMinedConfirmed =
           await walletClient.isTransactionMinedConfirmed(txIds[i]);
-        // let txnDetails = await wallet.getClient().getTransactionDetails(txIds[i]);
-        // consoleLogTransactionDetails(txnDetails, txIds[i]);
         expect(isTransactionMinedConfirmed).to.equal(true);
       }
     }
   }
 );
 
 When(
-  /I list all coinbase transactions for wallet (.*)/,
+  /I list all (.*) transactions for wallet (.*)/,
   { timeout: 20 * 1000 },
-  async function (walletName) {
+  async function (transaction_type, walletName) {
     const wallet = this.getWallet(walletName);
     const walletClient = await wallet.connectClient();
-    console.log("\nListing all coinbase transactions: ", walletName);
-    const transactions = await walletClient.getAllCoinbaseTransactions();
+    var transactions;
+    var type;
+    if (transaction_type === "NORMAL") {
+      transactions = await walletClient.getAllNormalTransactions();
+      type = "NORMAL";
```

Comment:
Nit: why declare a new variable for this when `transaction_type` is === to the "NORMAL"?

Reply:
This makes the step resilient. If the scenario states anything else than "NORMAL", it will be treated as "COINBASE". Variable `transaction_type` can be anything the user writes, but `type` is controlled and used in the logical branches.